modules/imgcodecs/src/grfmt_bmp.cpp

```diff
@@ -118,7 +118,7 @@ bool  BmpDecoder::readHeader()
 
                 if( m_bpp <= 8 )
                 {
-                    CV_Assert(clrused <= 256);
+                    CV_Assert(clrused >= 0 && clrused <= 256);
```

Comment:
This should work.

Anyway I believe more robust fix should do this:
```.diff
-int clrused
+unsigned int clrused
```

But probably this would require a lot of refactong because `getDWord()` method returns signed "int" (MSVS defined `DWORD` type as **unsigned** 32-bit integer). I/O API design is confusing here.